File: IntelliJFiles/src/ScoreCalc.java
import java.util.Scanner;

public class ScoreCalc {

    public static Scanner scanner = new Scanner(System.in);
    public static int matchNumber;
    public static String teamBlue;
    public static String teamRed;
    public static int BOn;
    public static int order;
    public static int bluePoints = 0;
    public static int redPoints = 0;
    public static String blueBan;
    public static String redBan;

    public static void main(String[] args) {

//        boolean finishMatch = false;

        System.out.println("Welcome to Speed Cup Score Calculator");

        setup();

        pickBan();

        ScoreCount();

        printResults();

//        while (!finishMatch) {
//            System.out.print("\nWhat would you like to do?" +
//                    "\n1. Setup" +
//                    "\n2. Pick Bans" +
//                    "\n3. Start The Match" +
//                    "\n\n> ");
//            int choice = scanner.nextInt();
//
//            switch (choice) {
//                case 1:
//                    setup();
//                    break;
//                case 2:
//                    pickBan();
//                    break;
//                case 3:
//                    ScoreCount();
//                    printResults();
//                    finishMatch = true;
//                    break;
//            }
//
//        }
    }

    public static void setup() {
        System.out.print("\nMatch ##?\n> ");
        matchNumber = scanner.nextInt();
        scanner.nextLine();

        System.out.print("\nWhat is the name of the blue team?\n> ");
        teamBlue = scanner.nextLine();

        System.out.print("\nWhat is the name of the red team?\n> ");
        teamRed = scanner.nextLine();

        System.out.print("\nBest of how many?\n> ");
        BOn = scanner.nextInt();
        scanner.nextLine();
        System.out.println("BO" + BOn + " = " + ((BOn / 2) + 1) + " points to win");
    }

    public static void pickBan() {

        System.out.print("\nWhich team won the roll?\n1. Blue  2. Red\n> ");
        order = scanner.nextInt();
        scanner.nextLine();

        if(order % 2 != 0) {
            System.out.print("\nTeam Red bans!\n\nWhat is Team Red's ban?\n> ");
            redBan = scanner.nextLine();

            System.out.print("\nTeam Blue bans!\n\nWhat is Team Blue's ban?\n> ");
            blueBan = scanner.nextLine();
        } else {
            System.out.print("\nTeam Blue bans!\n\nWhat is Team Blue's ban?\n> ");
            blueBan = scanner.nextLine();

            System.out.print("\nTeam Red bans!\n\nWhat is Team Red's ban?\n> ");
            redBan = scanner.nextLine();
        }
    }

    public static void ScoreCount() {

        while (bluePoints < ((BOn / 2) + 1) && redPoints < ((BOn / 2) + 1)) {

            if(order % 2 != 0) {
                System.out.println("\nTeam Blue's pick!");
            } else {
                System.out.println("\nTeam Red's pick!");
            }
            order++;

            System.out.print("\nTeam Blue's accuracy\n> ");
            double blueAcc = scanner.nextDouble();
            scanner.nextLine();

            System.out.print("\nTeam Red's accuracy\n> ");
            double redAcc = ((scanner.nextDouble() * 5) / 4);
            scanner.nextLine();

            System.out.println("red acc " + redAcc);

            if (blueAcc > redAcc) {
                bluePoints++;
            } else {
                redPoints++;
            }

            System.out.println("\n" + teamBlue + " " + bluePoints + " | " + redPoints + " " + teamRed);
        }

        if(bluePoints > redPoints) {
            System.out.println("\n" + teamBlue + " wins!");
        } else {
            System.out.println("\n" + teamRed + " wins!");
        }
    }

    public static void printResults() {
        System.out.print("\nMatch link\n> ");
        String matchLink = scanner.nextLine();

        System.out.println("\nResults:" +
                "\n\n__**Quarterfinals**__" + //change this for different weeks
                "\n**Match " + matchNumber + ":** " + matchLink +
                "\n\nResults: " + teamBlue + " **" + bluePoints + " | " + redPoints + "** " + teamRed +
                "\n\n**Bans:**" +
                "\n" + teamBlue + ": **" + blueBan + "**" +
                "\n" + teamRed + ": **" + redBan + "**");
    }


}

File: ScoreCalcPy/ScoreCalc.py
print("Welcome to Speed Cup Score Calculator")

matchNumber = int(input("\nMatch ##?\n> "))
teamBlue = input("\nWhat is the name of the blue team?\n> ")
teamRed = input("\nWhat is the name of the red team\n> ")
BOn = int(input("\nBest of how many?\n> "))
print(f"BO{BOn} = {BOn // 2 + 1} points to win")

input("\n\nWaiting for warmups to finish...")


order = int(input("\nWhich team won the roll?\n1. Blue 2. Red\n> "))

maps = ['DT1', 'DT2', 'DT3', 'DT4', 'DT5', 'DT6', 'DT7', 'DT8']
maps2 = ['DT9', 'DT10', 'DT11', 'DT12', 'DT13', 'NM1', 'NM2', 'TB']
mapIDs = ['124056', '715345', '1773874', '603965', '2036906', '207659', '263382', '507728']
mapIDs2 = ['2042978', '970516', '34162', '1236720', '152463', '2063785', '169841', '252238']

if order % 2 != 0:
    while True:
        redBan = input("\nTeam Red bans!\n\nWhat is Team Red's ban?\n> ")
        if redBan not in maps and redBan not in maps2:
            print("Pick a valid map")
            continue
        break
    while True:
        blueBan = input("\nTeam Blue bans!\n\nWhat is Team Blue's ban?\n> ")
        if blueBan not in maps and blueBan not in maps2:
            print("Pick a valid map")
            continue
        break
else:
    while True:
        blueBan = input("\nTeam Blue bans!\n\nWhat is Team Blue's ban?\n> ")
        if blueBan not in maps and blueBan not in maps2:
            print("Pick a valid map")
            continue
        break
    while True:
        redBan = input("\nTeam Red bans!\n\nWhat is Team Red's ban?\n> ")
        if redBan not in maps and redBan not in maps2:
            print("Pick a valid map")
            continue
        break


bluePoints = 0
redPoints = 0
playedMaps = []
isTied = False

while bluePoints < BOn // 2 + 1 and redPoints < BOn // 2 + 1:

    if order % 2 != 0:
        mapPick = input("\nTeam Blue's pick!\n> ")
    else:
        mapPick = input("\nTeam Red's pick!\n> ")
    while True:
        if mapPick in blueBan or mapPick in redBan:
            mapPick = input("Map banned\n> ")
        elif mapPick not in maps and mapPick not in maps2:
            mapPick = input("Pick a valid map\n> ")
        elif mapPick in playedMaps:
            mapPick = input("Map already played\n> ")
        elif mapPick in maps:
            print(f"!mp map {mapIDs[maps.index(mapPick)]}")
            break
        elif mapPick in maps2:
            print(f"!mp map {mapIDs2[maps2.index(mapPick)]}")
            break
    playedMaps.append(mapPick)
    order += 1

    input("\nWaiting for match to finish...")

    blueAcc = float(input("\nTeam Blue's accuracy\n> "))
    redAcc = float(input("\nTeam Red's accuracy\n> ")) * 5 / 4
    print(f"Red Acc: {redAcc:4}")

    if blueAcc > redAcc:
        bluePoints += 1
    else:
        redPoints += 1

    print(f"{teamBlue} {bluePoints} | {redPoints} {teamRed}")
    print("-------------------------------------------------")

    if bluePoints > redPoints:
        print(f"{teamBlue} wins!")
    elif bluePoints == BOn // 2 and redPoints == BOn // 2:
        print("\nTiebreaker time!")
        isTied = True
    else:
        print(f"{teamRed} wins!")

matchLink = input("\nMatch Link\n> ")

print(f"""__**Semifinals**__
**Match {matchNumber}:** {matchLink}

Results: {teamBlue} **{bluePoints} | {redPoints}** {teamRed}

**Bans:**
{teamBlue}: **{blueBan}**
{teamRed}: **{redBan}**""")


